tests/test_manageiq_provider.py

```diff
@@ -252,8 +261,28 @@ def test_generate_auth_key_config(miq):
                         'endpoint': {'hostname': PROVIDER_HOSTNAME,
                                      'port': PROVIDER_PORT,
                                      'role': 'default',
-                                     'certificate_authority': "",
-                                     'verify_ssl': PROVIDER_VERIFY_SSL}}
+                                     'certificate_authority': None,
+                                     'verify_ssl': PROVIDER_VERIFY_SSL,
+                                     'security_protocol': 'ssl-without-validation'}}
+
+
+def test_filter_empty_unrequired_fields(miq):
+    fields = {'field_that_should_be_ignored',
+              'field_that_should_be_sent',
+              'field_that_isnt_updated',
+              'empty_field_that_should_be_sent'}
+
+    configs = [{'endpoint': {'field_that_should_be_ignored': None,
+                             'field_that_should_be_sent': 'new_data',
+                             'empty_field_that_should_be_sent': None}}]
+
+    existing_endpoints = [{'field_that_isnt_updated': 'value',
+                           'field_that_should_be_sent': 'old_data',
+                           'empty_field_that_should_be_sent': 'some_data'}]
+
+    miq.filter_empty_unrequired_fields(configs, existing_endpoints, fields)
+    assert configs == [{'endpoint': {'field_that_should_be_sent': 'new_data',
+                                     'empty_field_that_should_be_sent': None}}]
```

Comment:
On one hand, awesome test, puts `fields` to neat use :heart: 
OTOH, it covers what the function *does* but it's hard to see *why* it achieves the desired backward-compat.
Perhaps scenario-like names like
```
unsupported_field_leaving_empty
unsupported_field_trying_to_set
supported_empty_leaving_empty
supported_empty_setting
supported_non_empty_clearing
supported_non_empty_setting
```
could help?
(The 3x2 matrix is bigger because some scenarios collapse to same function behavior.)